Tiltfile

```
@@ -388,6 +392,104 @@ def prepare_all():
     )
     local(cmd, env = settings.get("kustomize_substitutions", {}))
 
+# create cluster template resources from cluster-template files in the templates directory
+def cluster_templates():
+    substitutions = settings.get("kustomize_substitutions", {})
+
+    template_dirs = settings.get("template_dirs", {
+        "docker": ["./test/infrastructure/docker/templates"],
+    })
+
+    for provider, provider_dirs in template_dirs.items():
+        for template_dir in provider_dirs:
+            template_list = [filename for filename in listdir(template_dir) if os.path.basename(filename).endswith("yaml")]
+            for filename in template_list:
+                deploy_templates(filename, provider, substitutions)
+
+def deploy_templates(filename, provider, substitutions):
+    # validate filename exists
+    if not os.path.exists(filename):
+        fail(filename + " not found")
+
+    os.environ["NAMESPACE"] = substitutions.get("NAMESPACE", "default")
+    os.environ["KUBERNETES_VERSION"] = substitutions.get("KUBERNETES_VERSION", "v1.24.0")
+    os.environ["CONTROL_PLANE_MACHINE_COUNT"] = substitutions.get("CONTROL_PLANE_MACHINE_COUNT", "1")
+    os.environ["WORKER_MACHINE_COUNT"] = substitutions.get("WORKER_MACHINE_COUNT", "3")
+
+    basename = os.path.basename(filename)
+    if basename.endswith(".yaml"):
+        if basename.startswith("clusterclass-"):
+            template_name = basename.replace("clusterclass-", "").replace(".yaml", "")
+            deploy_clusterclass(template_name, provider, filename)
```

```suggestion
            clusterclass_name = basename.replace("clusterclass-", "").replace(".yaml", "")
            deploy_clusterclass(clusterclass_name, provider, filename)
```
nit (same in the func)